tabrepo/simulation/ensemble_selection_config_scorer.py

```diff
@@ -135,8 +148,69 @@ def evaluate_task(self, dataset: str, fold: int, models: List[str]) -> Tuple[flo
 
         ensemble_weights: np.array = weighted_ensemble.weights_
 
+        # ensemble_weights has to be updated, need to be in the original models order
+        ensemble_weights_fixed = np.zeros(n_models, dtype=np.float64)
+        ensemble_weights_fixed[models_filtered_idx] = ensemble_weights
+        ensemble_weights = ensemble_weights_fixed
+
         return err, ensemble_weights
 
+    def _get_models_filtered_idx(self, models: list[str], models_filtered: list[str]) -> Tuple[list[str], list[int]]:
+        """
+        Returns the filtered list of models and the index mapping of the filtered models to the original `models` list.
+        """
+        models_filtered_set = set(models_filtered)
+
+        # Preserve `models` order without duplicates
+        seen = set()
+        seen_add = seen.add
+        models_filtered = [m for m in models if (m in models_filtered_set) and not (m in seen or seen_add(m))]
```

Comment:
This line is a bit opaque. `(m in models_filtered_set)` is always true and `(m in seen or seen_add(m))` performs mutation in the for comprehension which makes it hard to understand. 

Could you write a comment explaining what is the operation done on `models_filtered`? It would probably makes sense to rename the variable here too, and use a for-loop to make the code easier to read.

Reply:
(m in models_filtered_set) is not always True, because `models_filtered_set` is a subset of `models`.

I've added comments and improved naming of variables. I've kept the list comprehension for compute efficiency.